pkg/apis/eventing/v1alpha1/subscription_validation.go

```diff
@@ -117,8 +116,23 @@ func isResultStrategyNilOrEmpty(r *ResultStrategy) bool {
 	return r == nil || equality.Semantic.DeepEqual(r, &ResultStrategy{}) || equality.Semantic.DeepEqual(r.Target, &corev1.ObjectReference{})
 }
 
-func isValidResultStrategy(r *ResultStrategy) *apis.FieldError {
-	return isValidObjectReference(*r.Target).ViaField("target")
+func isValidResultStrategy(r ResultStrategy) *apis.FieldError {
```

Comment:
FYI, I am taking some of these and moving it to a common place to reuse the logic: 

https://github.com/knative/eventing/pull/457/files#diff-96c67291a6d32860d990069d9d9d5cd3